src/main/java/com/endava/internship/collections/TreeIterator.java

```diff
@@ -0,0 +1,58 @@
+package com.endava.internship.collections;
+
+import java.util.Iterator;
+
+    public class TreeIterator implements Iterator<StudentSet.Node> {
+        private StudentSet.Node next;
```

Comment:
you can import StudentSet.Node in your TreeIterator class and use short name for Node